### PR TITLE
MGMT-12666: Document Unauthenticated Registries

### PR DESCRIPTION
**Pull request by CrystalChun**
https://issues.redhat.com/browse/MGMT-12666
Epic: https://issues.redhat.com/browse/MGMT-8789
Update documentation related to adding
`unauthenticatedRegistries` to the spec of the
`AgentServiceConfig` so users know how to specify
image registries that don't require authentication.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

/cc @carbonin

### CODE REVIEW

File: docs/operator.md [hunk 1]
```diff
@@ -333,3 +335,29 @@ EOF
 ```
 
 For more details on how to specify the CR, see [AgentServiceConfig CRD](https://github.com/openshift/assisted-service/blob/master/config/crd/bases/agent-install.openshift.io_agentserviceconfigs.yaml).
+
+### Image Registries Without Authentication
+
+`assisted-service` validates the [pull secret](hive-integration/kube-api-getting-started.md#2-create-a-pull-secret) provided for spoke cluster installation by making sure it contains the authentication information for every image registry used for installation. 
+
+Registries that don't require authentication can be added to an ignore list (called `PUBLIC_CONTAINER_REGISTRIES` in `assisted-service`). Any registry on the ignore list is not required to have an entry in the pull secret validated by the `assisted-service`.
```
Comment (danielerez) [inline docs/operator.md:343]: Isn't it sort of an implementation details at this point? I mean, maybe we can rephrase this to emphasize that the new API ('unauthenticatedRegistries') should be used. So in line 363, we can say that the 'ConfigMap override' is still supported for backwards compatibility. wdyt?

Comment (CrystalChun) [reply docs/operator.md]: Good point! I've reworded as suggested, how does it sound?

Comment (danielerez) [reply docs/operator.md]: looks good now:)
